### PR TITLE
implement click event on MapOverlay

### PR DESCRIPTION
**Pull request by ichan-mb**
Added support for clicked/tapped event on `MapOverlay`

**Examples:**
```
<App>
	<ClientPanel>
		<JavaScript>
			module.exports.overlayTapped = function(args) {
				console.log("overlay tapped");
			}
		</JavaScript>
		<NativeViewHost Dock="Top" Height="250">
			<MapView ux:Name="map" Zoom="14" Latitude="-6.914742" Longitude="107.609820">				
				<MapOverlay Radius="60" Type="Circle" FillColor="Red" Tapped="{overlayTapped}"
					CenterLatitude="-6.914742" CenterLongitude="107.609820" />
			</MapView>
		</NativeViewHost>
	</ClientPanel>
</App>
```

This PR contains:
- [ ] Changelog
- [ ] Documentation
- [ ] Tests

### CODE REVIEW

File: Source/Fuse.Maps/MapOverlay.uno [hunk 1]
```diff
@@ -115,6 +115,16 @@ namespace Fuse.Controls
 		static Selector _centerLatitudeName = "CenterLatitude";
 		static Selector _centerLongitudeName = "CenterLongitude";
 		static Selector _radiusName = "Radius";
+		static int UID_POOL = 0;
```
Comment (mortend) [inline Source/Fuse.Maps/MapOverlay.uno:118]: Code-style: Private fields should start with an underscore followed by a lower-case letter (`_uidPool`).

File: Source/Fuse.Maps/MapOverlay.uno [hunk 2]
```diff
@@ -115,6 +115,16 @@ namespace Fuse.Controls
...
+		internal int uid = UID_POOL++;
```
Comment (mortend) [inline Source/Fuse.Maps/MapOverlay.uno:119]: Code-style: Internal fields should start with an upper-case letter (`Uid`). Or perhaps just `Id`?

(It's unclear what the `U` means and `Id/id` seems to be used elsewhere.)

File: Source/Fuse.Maps/iOS/MapViewDelegate.m [hunk 3]
```diff
@@ -355,6 +360,30 @@ -(void)onTap:(UITapGestureRecognizer*)sender
...
+	{
+		if (overlaySelectBlock == NULL)
+			return;
+
+		MKMapPoint mappoint = MKMapPointForCoordinate(coord);
+		for (id<MKOverlay> overlay in _mapView.overlays)
+		{
+			MKOverlayPathRenderer * renderer = (MKOverlayPathRenderer*)[_mapView rendererForOverlay:overlay];
+			CGPoint tapPoint = [renderer pointForMapPoint:mappoint];
+			if (CGPathContainsPoint(renderer.path, nil, tapPoint, false))
```
Comment (mortend) [inline Source/Fuse.Maps/iOS/MapViewDelegate.m:377]: Nit: I would prefer to add curly-braces around the `if`-body when nested statements contain curly-braces, though not a big deal.

File: Source/Fuse.Maps/MapOverlay.uno (hunk 2 above)
Comment (ichan-mb) [reply Source/Fuse.Maps/MapOverlay.uno]: the logic here is that I borrowed it from the MapMarker, uid is used to identify which MapOverlay has been tapped

File: Source/Fuse.Maps/MapOverlay.uno (hunk 1 above)
Comment (ichan-mb) [reply Source/Fuse.Maps/MapOverlay.uno]: fixed

**Review by mortend — APPROVED** [review]
Thank you! :)

**Comment by mortend** [comment]
- [x] https://travis-ci.org/github/fuse-open/fuselibs/builds/753774128

**Pull request merged**